client/src/App.js

```diff
@@ -66,11 +72,59 @@ fetchFacilities = () => {
     this.render()
 }
 
+/////// Fetch Data ///////
+fetchFacilitiesSpecificMaterials = () => {
+  //Fetches facilities that can service user selcted materials.
+  if (this.state.selectedMaterials != null) {
+    let matString = "material_id[]=",
+        queryMats =[],
+        queryMatsString = '',
+        lat = this.state.userLat,
+        lng = this.state.userLng;
+
+    this.state.selectedMaterials.map(material=> {
+      queryMats.push(material.value);});
+      queryMatsString = queryMats.toString()
+
+    let url = `http://localhost:8080/api/facilities/earth911/facilities/search?lat=${lat}&lng=${lng}&materials=${queryMatsString}`;
+
+    fetch(url,{
+
+          })
+      .then((response) => {
+        return response.json();
+      })
+      .then((facilities_data) => {
+        this.setState({facilities:facilities_data.results.result});
+      });
+  } else {
+    this.setState({facilities:[]},  ()=>this.render())
+  }
+ }
+
+
+fetchMaterials = () => {
+  fetch('http://localhost:8080/api/facilities/earth911/materials')
+    .then((response) => {
+      return response.json();
+    })
+    .then((materials_data) => {
+      let materials = [];
+      materials_data.results.result.map(material=> {
+        materials.push({value:material.material_id,
+                        label:material.description})
+      });
+      this.setState({materials:materials})
+      });
+    this.render()
```

Comment:
`this.render()` comes after then() without then()? Or, was `this.render()` supposed to be a call back in setState()?

Reply:
Yeah, you are correct. Updated, along with a few others that were done like this.